### DISCUSSION TITLE
What is the right way to train on multiple datasets (for TTS)?

### DISCUSSION THREAD

**Discussion by iamanigeeit** [espnet/espnet#5253] [Q&A]
I'm surprised this hasn't been asked before.

Is it better to train sequentially, or put all the data in the same format and train together?

Problem with training sequentially: we need to know how many speakers in total, and the dataset stats are pre-calculated (?), which affect models that depend on these stats. For example, FS2 and JETS `pitch_embed` and `energy_embed` rely on the dataset pitch/energy range. When fine-tuning on a new dataset with bigger range, anything outside the original range is truncated.

Problem with combining datasets: data is not balanced (we have a lot of monotonous audiobook data but less expressive data). Can i just duplicate the smaller dataset multiple times to balance it?

**Comment by sw005320** [comment]
This is a good discussion.
If we want to make a generic system that covers all tasks and data, it would be better to start by combining the corpora.
You can check this project https://github.com/espnet/espnet/pull/5120
It's ASR, but anyway, we are adding over ten corpora, and the training data amounts to over 100K hours.

Yes, the data balancing is an issue, but we could upsample/downsample the data.
We could also tune the weight of the loss for this.
Larger model sizes can also mitigate the issue.
We're working on the above techniques and some of them would be released officially later.

If we want to focus on the performance of an individual database, we can fine-tune the generic model trained by the above step.

This is mostly for ASR, but some of them are common for TTS.
I want to have discussions from the others that have some experience with multiple datasets because our team is also struggling.

**Reply by iamanigeeit** [reply]
I will try combining datasets and if it works, maybe send in a PR after ASRU deadline. Thanks!

**Reply by sw005320** [reply]
Are you interested in our hackathon?
This will be a good topic.
If you're interested in it, please send me an email (shinjiw@ieee.org)
